src/components/new-safe/create/steps/ReviewStep/index.tsx

```diff
@@ -207,13 +209,14 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
           safeAccountConfig: props.safeAccountConfig,
           safeDeploymentConfig: {
             saltNonce,
-            safeVersion: LATEST_SAFE_VERSION as SafeVersion,
+            safeVersion: data.safeVersion,
           },
         },
       }
 
       const onSubmitCallback = async (taskId?: string, txHash?: string) => {
-        dispatch(addUndeployedSafe(undeployedSafe))
+        // Create a counterfactual Safe
+        await createCounterfactualSafe(chain, safeAddress, saltNonce, data, dispatch, props)
```

Comment:
```suggestion
        createCounterfactualSafe(chain, safeAddress, saltNonce, data, dispatch, props)
```

Now that it doesn't return a promise anymore.